linesearch.go

```diff
@@ -118,11 +103,7 @@ func (ls *LinesearchMethod) Iterate(loc *Location) (Operation, error) {
 		panic("linesearch: Linesearcher returned invalid operation")
 	}
 
-	if step == ls.lastStep {
-		// Linesearcher is requesting another evaluation at the same point
-		// which is stored in ls.loc.X.
-		copy(loc.X, ls.loc.X)
-	} else {
+	if step != ls.lastStep {
 		// We are moving to a new location.
```

Comment:
"We are moving to a new location, and not, say, evaluating extra information at the current location."